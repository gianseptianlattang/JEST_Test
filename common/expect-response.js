const { matchers } = require('jest-json-schema');
expect.extend(matchers);

export const expectStatus = (response, expectedStatus) => {
  expect((response).status).toEqual(expectedStatus);
};

export const expectSchema = (response, expectedSchema) => {
  expect((response).body).toMatchSchema(expectedSchema);
};

export const expectObject = (response, expectedObject) => {
  expect((response).body).toMatchObject(expectedObject);
};

export const expectSnapshot = (response) => {
  expect((response).body).toMatchSnapshot();
};