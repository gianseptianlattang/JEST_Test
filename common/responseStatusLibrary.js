export default {
  statusOk: 200,
  statusCreated: 201,
  statusNoContent: 204,
  statusBadRequest: 400,
  statusUnauthorized: 401,
  statusForbidden: 403,
  statusNotFound: 404,
  statusToLarge: 413,
  statusUnprocessableEntity: 422,
  statusInternalServerError: 500,
};
