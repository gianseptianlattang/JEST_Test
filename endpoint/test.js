const supertest = require("supertest");
require("dotenv").config();

const api = supertest("https://apingweb.com/api");

export const register = (body) =>
  api
    .post("/register")
    .set("Content-Type", "application/json")
    .set("Accept", "application/json")
    .send(body);

export const login = (body) =>
  api
    .post("/login")
    .set("Content-Type", "application/json")
    .set("Accept", "application/json")
    .send(body);

export const forgotPassword = (body) =>
  api
    .post("/forgot-password")
    .set("Content-Type", "application/json")
    .set("Accept", "application/json")
    .send(body);

export const resetPassword = (body) =>
  api
    .post("/reset-password")
    .set("Content-Type", "application/json")
    .set("Accept", "application/json")
    .send(body);
