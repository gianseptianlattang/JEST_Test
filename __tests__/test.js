import * as endpoint from "../endpoint/test";
import * as er from "../common/expect-response";
import rs from "../common/responseStatusLibrary";
import {
  bodyRegister,
  bodyLogin,
  bodyForgotPassword,
  bodyResetPassword,
} from "../data/data";

let response;
let body;
let email;
let token;

describe("Testing https://apingweb.com/#bearer", () => {
  test(`As a User, I can Create New Data`, async () => {
    body = bodyRegister();
    email = body.email;
    response = await endpoint.register(body);
    er.expectStatus(response, rs.statusOk);
    er.expectSnapshot(response);
  });

  test(`As a User, I can Login successfully`, async () => {
    response = await endpoint.login(bodyLogin(email));
    er.expectStatus(response, rs.statusOk);
    er.expectSnapshot(response);
  });

  test(`As a User, I can send email for Forgot Password successfully`, async () => {
    response = await endpoint.forgotPassword(bodyForgotPassword(email));
    token = response.body.or_use_this_token;
    er.expectStatus(response, rs.statusOk);
    er.expectSnapshot(response);
  });

  test(`As a User, I can Reset Password successfully`, async () => {
    response = await endpoint.resetPassword(bodyResetPassword(email, token));
    er.expectStatus(response, rs.statusOk);
    er.expectSnapshot(response);
  });
});
