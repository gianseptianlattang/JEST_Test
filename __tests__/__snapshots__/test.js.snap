// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing https://apingweb.com/#bearer As a User, I can Create New Data 1`] = `
Object {
  "errors": Array [],
  "message": "Registration success",
  "status": 200,
  "success": true,
  "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJhdWQiOiIxIiwianRpIjoiZTQzNzllMmRjNjZiMDk5MDdkN2IxZTZiNzg0NDBlZGNkYjFiNmYwMzk0MmU5YTdlNjNmZmYzMWY4ZDU5MmI2NTAzMzhiYjZkM2VlYmJiMWYiLCJpYXQiOjE2OTYyMTk2MTguODI2MzY3LCJuYmYiOjE2OTYyMTk2MTguODI2MzY5LCJleHAiOjE3Mjc4NDIwMTguODI0Niwic3ViIjoiMTQyNSIsInNjb3BlcyI6W119.XjkQnOkV3YTaKE_jk5SfShY9iz6rCUvfAqXoShkW-eZ91VuS0ed_OE1nMc6jMxz6tpcVClDttrgkWTvbwWgF9OwXEv3C7HU0VOSes5Lgxw5XgI_yrMdaZOeCmJXiZV-OdQcGqtj3y-JGDmmwoOo-7hOcKKZaEVlXB--LTHW2Q-YBDpvFISVK6vsnosl8cF5BDWRHluV2gnhHb0qlXOEcovjxF5fKLk-EtD25D4FTLBjTSZOMhGHijhu0LyjDM4z9IBYTXQ3nXUk5e27QSXQ1Kr56nGr0hrEGIHYA1QJw565Qkjv6IAmzp81XL2Ycg4bJzXu-lphJSTm1OAR93ffs3bwEcKLFBIwRlS0jqobiDauI0wWBK55b2neiXd4UjYuIva-j5rRCuWS1v5Jj9VwfOLA0XQ6Guej9rMKnnG2xF4dk4iFSrE74bMZztHvyAzQaiS4DmjsKQoXprDRE4eSaLWqnr6GBTZy_yDVAg6tNeTPyHDbduxsJy5L7T2Ju90toVSuDb8g7fRbY6L-MYlulSYg743YJL5qx5VHx29_JoYmOTNF3gY9wHXYODOAXsnZezH1hUWxHCV4hzNlHznnEwnXh80nAQJQ-W6FlpuTK0xy94S3vu-yiYIPYzLgb6JDgNwPsQeh6tXas9lV3H0GQK_yk5nSsVhKZ0TGeCLf7WLY",
}
`;

exports[`Testing https://apingweb.com/#bearer As a User, I can Login successfully 1`] = `
Object {
  "errors": Array [],
  "message": "Login success",
  "result": Object {
    "created_at": "2023-10-02T04:06:58.000000Z",
    "email": "superman644@mailsac.com",
    "email_verified_at": null,
    "id": 1425,
    "name": "Clark Kent",
    "phone": "5555551234",
    "updated_at": "2023-10-02T04:06:58.000000Z",
  },
  "status": 200,
  "success": true,
  "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJhdWQiOiIxIiwianRpIjoiOGFjZTBmNDM4Y2JlNzA1NTk4OGVlMWUxNzMyZmMzMWM0MWQ1NWNiZTU3ZmQ5MDYzN2I2NWM5MzI2NjhkYTRiZmY3YWQyYzRlMDJmY2ViZTYiLCJpYXQiOjE2OTYyMTk2MTkuODI1OTI5LCJuYmYiOjE2OTYyMTk2MTkuODI1OTMsImV4cCI6MTcyNzg0MjAxOS44MjM5OTUsInN1YiI6IjE0MjUiLCJzY29wZXMiOltdfQ.qMpYRRctrnZ3xVcQaHA9HQomM8-3iA6ELi5Mbs8sMlE9jRwGTRqmgOBruUa0ELUas9_pL2m4GB3xUn4LGEpdHg1LNyPgLCdCLOkRHfvmWO_oZOl-aXLIdHwL1aFmaiu8QziE5jlm75Eq-qzL0JLxxh41Q3DTq9S6isLda1NZPCzWU-5LVWKTS81Y6hYQxmqYRsE0lmdwZeauAKXEwOOfC_JqtZe-Yoh9B1ebPGDQTNFiGcp4G8FRB7nBIdjJfPg1uJieZHWEV8ld_ReIg9sSUDYbP8MuJqo0p6B0nTPeHOdIIzlalyDpPmigzV00zaaJAsaeb4AD4bm-2np4mOIOsJiPIJGvj8pVPhw3Fy0oL_uSut8fjExe1ywUr8nnQYG4mo2-g8iY7VTNxBmbtpGTZXkuJTycfksyuGnTixLZGY1ewzGdD0qfLvvEYk4Xw6XRWsEaaKlazjHySA9RlmgyKLkhx2FjevBMeiJNlePLOQKuKCs0uzyEkIYdkdrDW0qxoDdUqaI9aHN6E_e6GGX5-mGvrvEoUdwkHVVgZ3zjx96aYRfFj8HlBY1AUOrojKD3Ycc6ehUAcGc5rjUwxnexqxD0Uc_Fc9EBcyuQ1igk5Pr3f-cS9t-U3f3-_PLjeBl5H8M2C5aKAWNhu7IbDWwqDJe9jB5_uJiRQeprs9SYDVM",
}
`;

exports[`Testing https://apingweb.com/#bearer As a User, I can Reset Password successfully 1`] = `
Object {
  "errors": Array [],
  "message": "Your password has been changed",
  "status": 200,
  "success": true,
}
`;

exports[`Testing https://apingweb.com/#bearer As a User, I can send email for Forgot Password successfully 1`] = `
Object {
  "errors": Array [],
  "message": "We have e-mailed your password reset link, check your inbox or spam",
  "or_use_this_token": "kbLSpUhhJLUuwpJehqhfvTwvjBzMm5tcEfAnOlIHym9QhVAYnsQPNjRjfqmJ666p",
  "status": 200,
  "success": true,
}
`;
